card/mod/01_core/set/all/rules.rb

```diff
@@ -1,25 +1,41 @@
 RuleSQL = %{
   select rules.id as rule_id, settings.id as setting_id, sets.id as set_id, sets.left_id as anchor_id, sets.right_id as set_tag_id
-  from cards rules join cards sets on rules.left_id = sets.id join cards settings on rules.right_id = settings.id
+  from cards rules 
+  join cards sets     on rules.left_id  = sets.id 
+  join cards settings on rules.right_id = settings.id
   where sets.type_id     = #{Card::SetID }    and sets.trash     is false
   and   settings.type_id = #{Card::SettingID} and settings.trash is false
-  and                                             rules.trash    is false;
+  and                                             rules.trash    is false
+  and   (settings.codename != 'follow' or rules.db_content != '');
```

Comment:
I think you can use NOT (... and...), or for an exercise, express with AR relations and joins :)